src/views/VaultDetails.tsx

```diff
@@ -0,0 +1,258 @@
+import { shallowEqual } from 'react-redux';
+import { Link } from 'react-router-dom';
+import styled, { css } from 'styled-components';
+
+import { ButtonAction, ButtonShape, ButtonSize, ButtonType } from '@/constants/buttons';
+import { STRING_KEYS } from '@/constants/localization';
+import { NumberSign, TOKEN_DECIMALS, USD_DECIMALS } from '@/constants/numbers';
+import { AppRoute } from '@/constants/routes';
+
+import { useStringGetter } from '@/hooks/useStringGetter';
+import { useURLConfigs } from '@/hooks/useURLConfigs';
+
+import breakpoints from '@/styles/breakpoints';
+import { layoutMixins } from '@/styles/layoutMixins';
+
+import { AssetIcon } from '@/components/AssetIcon';
+import { Button } from '@/components/Button';
+import { Details } from '@/components/Details';
+import { Icon, IconName } from '@/components/Icon';
+import { Output, OutputType, ShowSign } from '@/components/Output';
+
+import { useAppSelector } from '@/state/appTypes';
+import { getCurrentMarketAssetData } from '@/state/assetsSelectors';
+import { getCurrentMarketData } from '@/state/perpetualsSelectors';
+import {
+  getCurrentMarketVaultDetails,
+  getCurrentMarketVaultMetadata,
+} from '@/state/vaultSelectors';
+
+import { getNumberSign } from '@/lib/numbers';
+
+export const VaultDetails: React.FC = () => {
+  const stringGetter = useStringGetter();
+  const { market, configs } = useAppSelector(getCurrentMarketData, shallowEqual) ?? {};
+  const { id, name } = useAppSelector(getCurrentMarketAssetData, shallowEqual) ?? {};
+  const { allTimePnl, currentLeverageMultiple, thirtyDayReturnPercent, currentPosition } =
+    useAppSelector(getCurrentMarketVaultDetails) ?? {};
+  const { totalValue } = useAppSelector(getCurrentMarketVaultMetadata) ?? {};
+  const { vaultsLearnMore } = useURLConfigs();
+
+  const items = [
+    {
+      key: 'balance',
+      label: stringGetter({ key: STRING_KEYS.VAULT_BALANCE }),
+      value: <Output value={totalValue} type={OutputType.CompactFiat} />,
+    },
+    {
+      key: 'all-time-pnl',
+      label: stringGetter({ key: STRING_KEYS.VAULT_ALL_TIME_PNL }),
+      value: (
+        <$ColoredReturn $sign={getNumberSign(allTimePnl?.absolute)}>
+          {allTimePnl?.absolute != null ? (
+            <Output value={allTimePnl?.absolute} type={OutputType.CompactFiat} />
+          ) : (
+            '-'
```

Comment:
does the '-' come for free if `allTimePnl?.absolute` is null/undefined?

Reply:
Oh yes it does do this for numbers (but not dates iirc)

Reply:
aaand nope, compact fiat throws an error if value is undefined. This component is rather inconsistent.